components/vc/front/src/cockpitLights.c

```diff
@@ -0,0 +1,111 @@
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+struct
+{
+    IMD_LIGHT_state_E imd_state;
+    BMS_LIGHT_state_E bms_state;
+} cockpitLights_data;
+
+/******************************************************************************
+*                       P U B L I C  F U N C T I O N S
+******************************************************************************/
+
+IMD_LIGHT_state_E IMD_LIGHT_getState(void)
+{
+    return cockpitLights_data.imd_state;
+}
+
+BMS_LIGHT_state_E BMS_LIGHT_getState(void)
+{
+    return cockpitLights_data.bms_state;
+}
+
+CAN_digitalStatus_E cockpitLights_imd_getStateCAN(void)
+{
+    return (cockpitLights_data.imd_state == IMD_LIGHT_ON) ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+CAN_digitalStatus_E cockpitLights_bms_getStateCAN(void)
+{
+    return (cockpitLights_data.bms_state == BMS_LIGHT_ON) ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+static void cockpitLight_init(void)
+{
+    cockpitLights_data.imd_state = IMD_LIGHT_OFF,
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+
+    cockpitLights_data.bms_state = BMS_LIGHT_OFF,
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_INACTIVE);
+}
+
+static void cockpitLights_periodic_10Hz(void)
+{
+    CAN_digitalStatus_E can_imd_status;
+    const bool IMD_LIGHT_valid = (CANRX_get_signal(VEH, BMSB_bmsStatusMem, &can_imd_status) == CANRX_MESSAGE_VALID);    
+
+    if (IMD_LIGHT_valid == true)
+    {
+        switch (cockpitLights_data.imd_state)
+        {
+            case IMD_LIGHT_ON:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+            case IMD_LIGHT_OFF:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+            default:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+        }
+    }
+
+    CAN_digitalStatus_E can_bms_status;
+    const bool BMS_LIGHT_valid = (CANRX_get_signal(VEH, BMSB_bmsStatusMem, &can_bms_status) == CANRX_MESSAGE_VALID);
+    
+    if (BMS_LIGHT_valid == true)
+    {
+        switch (cockpitLights_data.bms_state) 
+        {
+            case BMS_LIGHT_ON:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+            case BMS_LIGHT_OFF:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+            default:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+        }
+    }  
```

Comment:
Same thing here

Reply:
Also both of these you need to consider the case that the signal is invalid. If it is invalid, the lights should be on